### DISCUSSION TITLE
How to manually change reputation of a town?

### DISCUSSION THREAD

**Discussion by OverlordZorn** [Vdauphin/HeartsAndMinds#1397] [Question and answer] [answered]
Hi - if I understand right, each individual town has its own reputation towards the player faction, is that right?

Once the reputation is high enough, it wont be spawning the hostile faction, right?

We are currently running a campaign with a story teller heavy focus. Is there a way for a game master to manually change the reputation of a town with a script command?

**Comment by Vdauphin** [comment, marked as answer]
Hello,

> Hi - if I understand right, each individual town has its own reputation towards the player faction, is that right?

No, it is a global one

> Once the reputation is high enough, it wont be spawning the hostile faction, right?

No, enemy hostile spawn doesn't rely on reputation
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/activate.sqf#L103-L120
The number of hostile is defined the first time you enter in a city. It is a mix between the type of city, the group ratio define in mission parameter (`Enemy density`) and the `Density of military in house` (if you select 100%, all enemy will be in house)
After that, when you enter again, enemies not killed previously will appear again. Basically, if a city looks empty it is because you clean it or it was a free city from the start (https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L134-L139)

> We are currently running a campaign with a story teller heavy focus. Is there a way for a game master to manually change the reputation of a town with a script command?

Change the value of `btc_global_reputation` server side but it kill the actual H&M gameplay

Cheers

**Reply by OverlordZorn** [reply]
Okey understood. I dont want to change the global reputation in this case 
My goal is mainly to control the "we liberated this town" "we set up a checkpoint, so these towns are not under the unfluence of the enemy anymore" - how can i ensure that this town actually is "liberated" and not spawn enemies anymore? 
Purely story telling wise.

I noticed during testing with in debug in the editor, sometimes when entering a town, there is a group of hostiles spawned far away, that make their way towards the town. 
Are those volitile reinforcements or are those part of the "fixed, defined by entering a town for the first time" - assosiated garrison?

Is there a way to read out the enemy presence in a town?
What is an indicator of the town being fully liberated?

Furthermore: 
We had a situation where we patrolled a town and fully cleared out the enemy present. I observed via zeus, i couldnt find any units ether inside nor on their way to the town (i added all units present on the map to the curator etc).
Next day, after a restart, we went there again and we have encountered more enemies, eventhough we thought it would be fully clear from the day before. Therefore the assumption there is a local reputation per town kind of a deal that defines if "new enemies" appear or not.

**Reply by Vdauphin** [reply]
> how can i ensure that this town actually is "liberated" and not spawn enemies anymore?

City are liberated once you kill all enemies, after that enemies will not spawn. It is persistant, if you kill 5 enemies and you leave 5 enemies in the city they will stay here. If you kill all of them, the city will stay free of enemy. 
It is the same for civilian, if you kill them, they will not magically spawn again


> I noticed during testing with in debug in the editor, sometimes when entering a town, there is a group of hostiles spawned far away, that make their way towards the town.
Are those volitile reinforcements or are those part of the "fixed, defined by entering a town for the first time" - assosiated garrison?

There are two system allowing IA to spawn outside the city you activate. Patrol of enemy (similar as traffic for civilian), they are not persistant and they patrol across city and get remove if no player is around and spawn again later when palyer move around the map
The second is reinforcement from hideout, they are send from hideout to free city in a certain cercle around. Once they reach the free city, they are put in garrison, the city become occupied and all those guys are saved inside the city (persiste)

> Is there a way to read out the enemy presence in a town?

Talking to civilians or simply use darter drone to locate enemies

>  What is an indicator of the town being fully liberated?

No enemies inside and civilian will report enemies far from the city you are next to

> Next day, after a restart, we went there again and we have encountered more enemies, eventhough we thought it would be fully clear from the day before. 

May be you met a patrol or a hideout send reinforcement when you left the city (reinforcement are send when you left city to surprise the player once they come back https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/de_activate.sqf#L120)

> Therefore the assumption there is a local reputation per town kind of a deal that defines if "new enemies" appear or not.

http://vdauphin.github.io/HeartsAndMinds/InGame-documentation#hideout
Actually there is no point to clear city during the first phase of the mission

**Reply by OverlordZorn** [reply]
Thank you very much, that was very insightful!
